sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventService.java

```diff
@@ -528,6 +528,16 @@ public Predicate buildCriteriaFilter(EventCriteria eventCriteria, CriteriaBuilde
 
 			filter = and(cb, filter, cb.isFalse(eventParticipantJoin.get(EventParticipant.DELETED)));
 		}
+		if (eventCriteria.getPerson() != null) {
+			Join<Event, EventParticipant> eventParticipantJoin = from.join(Event.EVENT_PERSONS, JoinType.LEFT);
```

Comment:
I think it would be a good idea to do the join from event to event participant at the start of the method because it's also used when the event criteria has a case, like this:

```
Join<Event, EventParticipant> eventParticipantJoin = null;
if (eventCriteria.getCaze() != null || eventCriteria.getPerson() != null) {
  eventParticipantJoin = from.join(Event.EVENT_PERSONS, JoinType.LEFT);
}
```

That makes sure that we don't do the join twice which would probably slow down the query.

Reply:
This isn't the most sustainable thingy thing. While I was at it, I added an EventJoins class and refactored the uses of event joins.

Reply:
Great, thank you!